src/TensorFlowNET.Core/Variables/variables.py.cs

```diff
@@ -155,7 +157,10 @@ public static Operation _safe_initial_value_from_op(string name, Operation op, D
 
         public static Tensor global_variables_initializer()
         {
-            throw new NotImplementedException();
+            // if context.executing_eagerly():
+            //      return control_flow_ops.no_op(name = "global_variables_initializer")
+            var group =  variables_initializer(global_variables().ToArray());
```

Comment:
@Wanglongzhi2001 an implicit cast happens there. `group` is actually an `Operation`. `Tensor` constructor goes to `Operation._ouputs.FirstOrDefault()` and returns `null` since `_outputs` for `no_op` are empty. This `null` than goes to `evaluate` functions and causes `NullReferenceException` since `null` can't be evaluated. 
So, the question is "shouldn't we return some placeholder instead of null"? If not than how should we rewrite the `evalute` function to support `no_op`?

Reply:
According to the implemention in tensorflow, we should return placeholder:
https://github.com/tensorflow/tensorflow/blob/6334682d8fcd1413eab2464e35696be48638a7a7/tensorflow/python/framework/test_util.py#L2801-L2810

Reply:
I have the following exception when run on a placeholder Tensor: 
Tensorflow.InvalidArgumentError: 'You must feed a value for placeholder tensor 'Placeholder' with dtype float
	 [[{{node Placeholder}}]]'
	 
Should I choose a different dtype?

Reply:
Did you feed the placeholder with different type?

Reply:
It turned out I just had to run `session.run(variable)` that initializes it.